Yafc/Workspace/ProductionTable/ProductionLinkSummaryScreen.cs

```diff
@@ -114,6 +104,20 @@ private void ShowRelatedLinks(ImGui gui) {
         }
     }
 
+    public static void FindProductionTable(ModelObject link, out ModelObject table, out List<ModelObject> parents) {
```

Comment:
I'm hoping this method signature can be changed as follows, to match the second suggestion in my other comment. It will require adjusting the implementation, though.
```suggestion
    public static void FindProductionTable(ModelObject link, out ProductionTable table, out List<ModelObject> parents) {
```

Reply:
Because of the way YaFC builds up the object tree, it's always a project that contains project pages that contains a production table that contains rows with nested rows, recipes and links. What this function does is travel upwards in this tree until it finds the project page. It then returns the second last nesting, the production table. (Not sure if i hae all names here correct, but that's about the gist of it). All these objects extend ModelObject.

Unless we ever change the way this object tree is build in YaFC, this function should always return a ProductionTable, however the code cannot guarantee this here, so the compiler believes a null could come out. It's good style to handle this null properly.

I'll modify this function and the places that call it to have a more proper return type.